functions/src/definitions/eventHandlers/onInstanceUpdate.ts

```diff
@@ -4,30 +4,35 @@ import {
   CollectionTypes,
 } from "../common/typesense/collectionOperations"
 import { getEmbedding } from "../common/machineLearningServer/operations"
+import { onDocumentUpdated } from "firebase-functions/v2/firestore"
 
-const onInstanceUpdate = functions
-  .region("asia-southeast1")
-  .runWith({
+const onInstanceUpdateV2 = onDocumentUpdated(
+  {
+    document: "/messages/{messageId}/instances/{instanceId}",
     secrets: ["TYPESENSE_TOKEN", "ML_SERVER_TOKEN"],
-  })
-  .firestore.document("/messages/{messageId}/instances/{instanceId}")
-  .onUpdate(async (change, context) => {
+  },
+  async (event) => {
     // Grab the current value of what was written to Firestore.
-    const before = change.before.data()
-    const after = change.after.data()
+    if (!event?.data?.before || !event?.data?.after) {
+      return Promise.resolve()
+    }
+    const before = event.data.before.data()
+    const snap = event.data.after
```

Comment:
Can apply the more descriptive names here as well

Reply:
Ok done